dartagnan/src/main/java/com/dat3m/dartagnan/Dartagnan.java

```diff
@@ -215,6 +212,24 @@ public static void main(String[] args) throws Exception {
         }
     }
 
+    public static File generateExecutionGraphFile(VerificationTask task, ProverEnvironment prover, ModelChecker modelChecker,
+                                                  WitnessType witnessType)
+            throws InvalidConfigurationException, SolverException, IOException {
+        Preconditions.checkArgument(modelChecker.hasModel(), "No model to generate.");
+
+        final ExecutionModel m = ExecutionModel.withContext(modelChecker.getEncodingContext());
+        m.initialize(prover.getModel());
+        final SyntacticContextAnalysis synContext = newInstance(task.getProgram());
+        final String name = task.getProgram().getName().substring(0, task.getProgram().getName().lastIndexOf('.'));
```

Comment:
The error I see in the console is due to this line. Somehow, the program has no name

Reply:
Oh. You originally said that you got an error when running *from the console* which is what I couldn't reproduce.
You meant to say that you get an error when running *from the UI* I guess. That one I can reproduce.

Reply:
Sorry ... By console I meant the display from the UI where we are supposed to see the result

Reply:
Fixed